backend/coreapp/views/preset.py

```diff
@@ -13,31 +15,48 @@
 logger = logging.getLogger(__name__)
 
 
+class AuthorizationException(APIException):
+    status_code = status.HTTP_403_FORBIDDEN
+    default_detail = "You must be logged in to perform this action."
+
+
 class PresetPagination(CursorPagination):
     ordering = "-creation_time"
     page_size = 10
     page_size_query_param = "page_size"
     max_page_size = 100
 
 
-class ReadOnly(BasePermission):
-    def has_permission(self, request: Any, view: Any) -> bool:
-        return request.method in SAFE_METHODS
+class IsOwnerOrReadOnly(BasePermission):
+    def has_object_permission(self, request: Any, view: Any, obj: Any) -> bool:
+        if request.method in SAFE_METHODS:
+            return True
+        if isinstance(obj, Preset):
+            return obj.owner == request.profile
```

Comment:
`or if request.profile.is_staff()`

if my syntax was wrong, we do this for making sure staff can delete scratches, so you should be able to use the same pattern

Reply:
This is inside the `IsOwnerOrReadonly` and the permissiosn are defined as followed

https://github.com/decompme/decomp.me/blob/9bc7f8a073c81582ddbcb1d03fc83e9ce9a4bed9/backend/coreapp/views/preset.py#L40

Therefore not requiring admin to be checked. I can add a unit test to ensure the expected behaviour if necessary

Reply:
ahhh missed that! thanks :)